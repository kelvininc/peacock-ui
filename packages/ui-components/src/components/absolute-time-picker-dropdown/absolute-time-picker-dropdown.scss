@import '../../assets/styles/globals';

:host {
	display: block;
}

.absolute-time-content {
	background-color: kv-color('neutral-7');
	border: 1px solid kv-color('neutral-6');
	border-radius: 4px;
	width: max-content;

	.footer {
		display: flex;
		justify-content: flex-end;
		padding: $spacing-4x;
		border-top: 1px solid kv-color('neutral-6');
	
		.actions {
			display: flex;
			gap: $spacing-2x;
		}
	}	
}
