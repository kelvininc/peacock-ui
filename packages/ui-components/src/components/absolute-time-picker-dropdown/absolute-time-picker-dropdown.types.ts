import { EventEmitter } from '@stencil/core';
import { ITextField } from '../text-field/text-field.types';
import { EAbsoluteTimePickerMode, ETooltipPosition, ITimePickerTimezone, ITimezoneOffset, SelectedTimestamp } from '../../types';
import { ComputePositionConfig } from '@floating-ui/dom';

export interface IAbsoluteTimePickerDropdown {
	/** (optional) Defines if the calendar is in single date or range mode */
	mode?: EAbsoluteTimePickerMode;
	/** (optional) Enables the back button displayed on top */
	displayBackButton?: boolean;
	/** (optional) Selected dates */
	selectedDates?: SelectedTimestamp;
	/** (optional) Initial Date */
	initialDate?: string;
	/** (optional) Disabled dates */
	disabledDates?: string[];
	/** (optional) calendar minimum date to be navigated */
	calendarInputMinDate?: string;
	/** (optional) calendar maximum date to be navigated */
	calendarInputMaxDate?: string;
	/** (optional) dropdown input config */
	inputConfig?: Partial<ITextField>;
	/** (optional) Dropdown positioning config */
	dropdownPositionOptions?: Partial<ComputePositionConfig>;
	/** (optional) Positioning of the tooltip */
	tooltipPosition?: ETooltipPosition;
	/** (Optional) Timezone of the selected dates */
	timezone?: ITimePickerTimezone;
	/** (optional) Available Timezones */
	timezones?: ITimezoneOffset[];
	/** (optional) Title displayed on top of the component */
	headerTitle?: string;
}

export interface IAbsoluteTimePickerDropdownEvents {
	/** Selected dates change */
	selectedDatesChange: EventEmitter<SelectedTimestamp>;
	/** Cancel button clicked */
	cancelClicked: EventEmitter<CustomEvent<MouseEvent>>;
	/** Dropdown open state change */
	dropdownStateChange: EventEmitter<boolean>;
}
