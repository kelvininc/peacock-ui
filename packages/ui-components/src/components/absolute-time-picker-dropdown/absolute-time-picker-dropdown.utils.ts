import { isEmpty, isNil, isNumber } from 'lodash-es';
import { EAbsoluteTimeError, EAbsoluteTimePickerMode, IAbsoluteTimeLimits, SelectedRange, SelectedTimestamp } from '../../types';
import { CALENDAR_DATE_TIME_MASK } from '../absolute-time-picker/absolute-time-picker.config';
import { createTimestampInTimezoneFromFormattedDate } from '../time-picker/time-picker.helper';
import dayjs from 'dayjs';

export const getSelectedTimestampDates = (range: SelectedRange, mode: EAbsoluteTimePickerMode, timezoneName: string): SelectedTimestamp => {
	if (range?.length === 0) return [];

	if (mode === EAbsoluteTimePickerMode.Single) {
		const [date] = range;

		return [createTimestampInTimezoneFromFormattedDate(date, timezoneName, CALENDAR_DATE_TIME_MASK)];
	}

	const [from, to] = range;
	const dateFrom = createTimestampInTimezoneFromFormattedDate(from, timezoneName, CALENDAR_DATE_TIME_MASK);
	const dateTo = !isEmpty(to) ? createTimestampInTimezoneFromFormattedDate(to, timezoneName, CALENDAR_DATE_TIME_MASK) : undefined;

	return [dateFrom, dateTo];
};

export const getFormattedSelectedDates = (range: SelectedTimestamp, mode: EAbsoluteTimePickerMode, timezoneName: string): SelectedRange => {
	if (isEmpty(range)) return [];

	if (mode === EAbsoluteTimePickerMode.Single) {
		const [date] = range;

		return [dayjs(date).tz(timezoneName).format(CALENDAR_DATE_TIME_MASK)];
	}

	const [from, to] = range;
	const dateFrom = dayjs(from).tz(timezoneName).format(CALENDAR_DATE_TIME_MASK);
	const dateTo = !isNil(to) ? dayjs(to).tz(timezoneName).format(CALENDAR_DATE_TIME_MASK) : undefined;

	return [dateFrom, dateTo];
};

export const getAbsoluteTimePickerError = (range: SelectedTimestamp, mode: EAbsoluteTimePickerMode, limits: IAbsoluteTimeLimits): EAbsoluteTimeError | undefined => {
	if (!isAbsoluteTimePickerFilled(range, mode)) {
		return;
	}

	if (mode === EAbsoluteTimePickerMode.Single) {
		const [date] = range;

		if (limits.minDate && dayjs(date).isBefore(limits.minDate)) {
			return EAbsoluteTimeError.StartDateBeforeMinimumDate;
		}

		if (limits.maxDate && dayjs(date).isAfter(limits.maxDate)) {
			return EAbsoluteTimeError.EndDateAfterMaximumDate;
		}

		return;
	}

	const [startDate, endDate] = range;

	if (!dayjs(endDate).isAfter(startDate)) {
		return EAbsoluteTimeError.EndDateBeforeStartDate;
	}

	if (limits.minDate && dayjs(startDate).isBefore(limits.minDate)) {
		return EAbsoluteTimeError.StartDateBeforeMinimumDate;
	}

	if (limits.maxDate && dayjs(endDate).isAfter(limits.maxDate)) {
		return EAbsoluteTimeError.EndDateAfterMaximumDate;
	}

	return;
};

export const isAbsoluteTimePickerFilled = (range: SelectedTimestamp, mode: EAbsoluteTimePickerMode): boolean => {
	if (isEmpty(range)) return false;

	if (mode === EAbsoluteTimePickerMode.Single) {
		const [date] = range;

		return isNumber(date);
	}

	const [from, to] = range;

	return isNumber(from) && isNumber(to);
};
