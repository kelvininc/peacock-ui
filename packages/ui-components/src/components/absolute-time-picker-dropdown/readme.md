# absolute-time-picker-dropdown



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Default -->
<kv-absolute-time-picker-dropdown></kv-absolute-time-picker-dropdown>

<!-- With selected ranges date -->
<kv-absolute-time-picker-dropdown [selectedDates]="['2022-08-04', '2022-09-01']"></kv-absolute-time-picker-dropdown>

<!-- With initial date -->
<kv-absolute-time-picker-dropdown initialDate="2022-08-04"></kv-absolute-time-picker-dropdown>

<!-- With disabled dates-->
<kv-absolute-time-picker-dropdown [disabledDates]="['2021-12-01', '2021-12-04', '2021-12-05']"></kv-absolute-time-picker-dropdown>

<!-- With min and max date -->
<kv-absolute-time-picker-dropdown minDate="2021-12-04" maxDate="2023-02-12"></kv-absolute-time-picker-dropdown>
```


### React

```tsx
import React from 'react';

import { KvAbsoluteTimePickerDropdown } from '@kelvininc/react-ui-components';

export const KvAbsoluteTimePicker: React.FC = () => (
	<>
		{/*-- Default --*/}
		<KvAbsoluteTimePickerDropdown />

		{/*-- With selected date --*/}
		<KvAbsoluteTimePickerDropdown selectedDates={['2022-08-04', '2022-09-01']} />

		{/*-- With initial date --*/}
		<KvAbsoluteTimePickerDropdown initialDate="2021-12-03" />

		{/*-- With disabled dates --*/}
		<KvAbsoluteTimePickerDropdown disabledDates={['2021-12-01', '2021-12-04', '2021-12-05']} />

		{/*-- With min and max --*/}
		<KvAbsoluteTimePickerDropdown minDate="2021-12-04" maxDate="2023-02-12" />
	</>
);
```



## Properties

| Property                  | Attribute                 | Description                                                        | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | Default                                        |
| ------------------------- | ------------------------- | ------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------- |
| `calendarInputMaxDate`    | `calendar-input-max-date` | (optional) calendar maximum date to be navigated                   | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `undefined`                                    |
| `calendarInputMinDate`    | `calendar-input-min-date` | (optional) calendar minimum date to be navigated                   | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `undefined`                                    |
| `disabledDates`           | --                        | (optional) Disabled dates                                          | `string[]`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | `[]`                                           |
| `dropdownPositionOptions` | --                        | (optional) Dropdown positioning config                             | `{ placement?: Placement; strategy?: Strategy; middleware?: (false \| { name: string; options?: any; fn: (state: { platform: Platform; placement: Placement; strategy: Strategy; x: number; y: number; initialPlacement: Placement; middlewareData: MiddlewareData; rects: ElementRects; elements: Elements; }) => Promisable<MiddlewareReturn>; })[]; platform?: Platform; }`                                                                                                                                                                                                                                                                                              | `DEFAULT_TIME_RANGE_DROPDOWN_POSITION_OPTIONS` |
| `headerTitle`             | `header-title`            | (optional) Title displayed on top of the component                 | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `''`                                           |
| `initialDate`             | `initial-date`            | (optional) Initial Date                                            | `string`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `undefined`                                    |
| `inputConfig`             | --                        | (optional) dropdown input config                                   | `{ type?: EInputFieldType; label?: string; icon?: EIconName \| EOtherIconName; actionIcon?: EIconName \| EOtherIconName; inputName?: string; examples?: string[]; placeholder?: string; maxLength?: number; minLength?: number; max?: string \| number; min?: string \| number; step?: string \| number; size?: EComponentSize; disabled?: boolean; required?: boolean; loading?: boolean; state?: EValidationState; helpText?: string \| string[]; value?: string \| number; valuePrefix?: string; badge?: string; readonly?: boolean; forcedFocus?: boolean; tooltipConfig?: Partial<ITooltip>; useInputMask?: boolean; inputMaskRegex?: string; fitContent?: boolean; }` | `{}`                                           |
| `mode`                    | `mode`                    | (optional) Defines if the calendar is in single date or range mode | `EAbsoluteTimePickerMode.Range \| EAbsoluteTimePickerMode.Single`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | `EAbsoluteTimePickerMode.Single`               |
| `selectedDates`           | --                        | (optional) Selected dates                                          | `[] \| [number, number] \| [number]`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | `[]`                                           |
| `timezone`                | --                        | (Optional) Timezone of the selected dates                          | `{ name: string; offset: number; }`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | `undefined`                                    |
| `timezones`               | --                        | (optional) Available Timezones                                     | `ITimezoneOffset[]`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | `buildTimezoneByOffset(getTimezonesNames())`   |
| `tooltipPosition`         | `tooltip-position`        | (optional) Positioning of the tooltip                              | `ETooltipPosition.Bottom \| ETooltipPosition.BottomEnd \| ETooltipPosition.BottomStart \| ETooltipPosition.Left \| ETooltipPosition.LeftEnd \| ETooltipPosition.LeftStart \| ETooltipPosition.Right \| ETooltipPosition.RightEnd \| ETooltipPosition.RightStart \| ETooltipPosition.Top \| ETooltipPosition.TopEnd \| ETooltipPosition.TopStart`                                                                                                                                                                                                                                                                                                                            | `ETooltipPosition.BottomStart`                 |


## Events

| Event                 | Description                | Type                                              |
| --------------------- | -------------------------- | ------------------------------------------------- |
| `cancelClicked`       | Cancel button clicked      | `CustomEvent<CustomEvent<MouseEvent>>`            |
| `dropdownStateChange` | Dropdown open state change | `CustomEvent<boolean>`                            |
| `selectedDatesChange` | Selected dates change      | `CustomEvent<[] \| [number, number] \| [number]>` |


## Dependencies

### Depends on

- [kv-dropdown](../dropdown)
- [kv-absolute-time-picker](../absolute-time-picker)
- [kv-action-button-text](../action-button-text)

### Graph
```mermaid
graph TD;
  kv-absolute-time-picker-dropdown --> kv-dropdown
  kv-absolute-time-picker-dropdown --> kv-absolute-time-picker
  kv-absolute-time-picker-dropdown --> kv-action-button-text
  kv-dropdown --> kv-dropdown-base
  kv-dropdown --> kv-text-field
  kv-dropdown-base --> kv-portal
  kv-text-field --> kv-tooltip
  kv-text-field --> kv-form-label
  kv-text-field --> kv-icon
  kv-text-field --> kv-badge
  kv-text-field --> kv-form-help-text
  kv-tooltip --> kv-portal
  kv-tooltip --> kv-tooltip-text
  kv-form-help-text --> kv-icon
  kv-absolute-time-picker --> kv-icon
  kv-absolute-time-picker --> kv-date-time-input
  kv-absolute-time-picker --> kv-time-picker-calendar
  kv-date-time-input --> kv-form-label
  kv-time-picker-calendar --> kv-icon
  kv-time-picker-calendar --> kv-time-picker-calendar-day
  kv-action-button-text --> kv-action-button
  kv-action-button-text --> kv-icon
  style kv-absolute-time-picker-dropdown fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


