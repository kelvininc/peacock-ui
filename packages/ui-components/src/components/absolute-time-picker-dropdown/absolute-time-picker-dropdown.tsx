import { Component, Event, EventEmitter, Host, Prop, State, Watch, h } from '@stencil/core';
import { EAbsoluteTimePickerMode, IAbsoluteSelectedRangeDates } from '../absolute-time-picker/absolute-time-picker.types';
import { getDefaultTimezone, getTimezoneOffset, getTimezonesNames } from '../../utils/date.helper';
import { CALENDAR_MASK } from '../absolute-time-picker/absolute-time-picker.config';
import { buildCustomIntervalTimeRange, buildTooltipText, hasRangeChanged } from '../time-picker/time-picker.helper';
import { SelectedTimestamp } from '../time-picker/time-picker.types';
import { isEmpty, merge } from 'lodash';
import { ITextField } from '../text-field/text-field.types';
import { DEFAULT_TIME_RANGE_DROPDOWN_POSITION_OPTIONS, DEFAULT_TIME_RANGE_PICKER_INPUT_CONFIG } from '../time-picker/time-picker.config';
import { ComputePositionConfig } from '@floating-ui/dom';
import dayjs from 'dayjs';
import { EActionButtonType } from '../action-button/action-button.types';
import { EComponentSize, ETooltipPosition, ITimePickerTimezone, ITimezoneOffset } from '../../types';
import { IAbsoluteTimePickerDropdown, IAbsoluteTimePickerDropdownEvents } from './absolute-time-picker-dropdown.types';
import { buildTimezoneByOffset } from '../calendar-advanced-date-selector/calendar-advanced-date-selector.helper';
import { getSelectedTimestampDates, getFormattedSelectedDates } from './absolute-time-picker-dropdown.utils';

@Component({
	tag: 'kv-absolute-time-picker-dropdown',
	styleUrl: 'absolute-time-picker-dropdown.scss',
	shadow: false
})
export class KvAbsoluteTimePickerDropdown implements IAbsoluteTimePickerDropdown, IAbsoluteTimePickerDropdownEvents {
	/** @inheritdoc */
	@Prop({ reflect: false }) inputConfig?: Partial<ITextField> = {};
	/** @inheritdoc */
	@Prop({ reflect: false }) dropdownPositionOptions?: Partial<ComputePositionConfig> = DEFAULT_TIME_RANGE_DROPDOWN_POSITION_OPTIONS;
	/** @inheritdoc */
	@Prop({ reflect: false }) selectedDates?: SelectedTimestamp = [];
	/** @inheritdoc */
	@Prop({ reflect: false }) timezone?: ITimePickerTimezone;
	/** @inheritdoc */
	@Prop({ reflect: false }) timezones?: ITimezoneOffset[] = buildTimezoneByOffset(getTimezonesNames());
	/** @inheritdoc */
	@Prop({ reflect: false }) disabledDates?: string[] = [];
	/** @inheritdoc */
	@Prop({ reflect: false }) initialDate?: string;
	/** @inheritdoc */
	@Prop({ reflect: false }) calendarInputMinDate?: string;
	/** @inheritdoc */
	@Prop({ reflect: false }) calendarInputMaxDate?: string;
	/** @inheritdoc */
	@Prop({ reflect: false }) mode?: EAbsoluteTimePickerMode = EAbsoluteTimePickerMode.Single;
	/** @inheritdoc */
	@Prop({ reflect: false }) tooltipPosition?: ETooltipPosition = ETooltipPosition.BottomStart;
	/** @inheritdoc */
	@Prop({ reflect: false }) headerTitle?: string = '';

	// Defines the calendar initial date if needed
	@State() calendarInitialDate: string = this.initialDate;
	// Current selected option
	@State() selectedDateState: SelectedTimestamp = [];
	// Dropdown open state
	@State() dropdownOpen: boolean = false;

	/** @inheritdoc */
	@Event() selectedDatesChange: EventEmitter<SelectedTimestamp>;
	/** @inheritdoc */
	@Event() cancelClicked: EventEmitter<CustomEvent<MouseEvent>>;
	/** @inheritdoc */
	@Event() dropdownStateChange: EventEmitter<boolean>;

	componentWillLoad() {
		if (this.selectedDates?.length > 0 && this.selectedDateState?.length === 0) {
			this.selectedDateState = this.selectedDates;
			this.handleSelecteDatesChange(this.selectedDates);
		}
	}

	@Watch('selectedDates')
	handleSelecteDatesChange(newDates: SelectedTimestamp) {
		this.selectedDateState = newDates;
		if (isEmpty(this.initialDate) && newDates?.length > 0) {
			const [fromDate] = newDates;
			this.calendarInitialDate = dayjs(fromDate).format(CALENDAR_MASK);
		}
	}

	private getSelectedTimezone = (): ITimePickerTimezone => {
		if (!isEmpty(this.timezone)) return this.timezone;

		const defaultTimezone = getDefaultTimezone();
		return {
			name: defaultTimezone,
			offset: getTimezoneOffset(defaultTimezone)
		};
	};

	private handleAbsoluteDatesChange = ({ detail }: CustomEvent<IAbsoluteSelectedRangeDates>) => {
		const range = detail.range;
		const { name: timezoneName } = this.getSelectedTimezone();
		this.selectedDateState = getSelectedTimestampDates(range, this.mode, timezoneName);
	};

	private getAbsoluteRange = (): string[] => {
		const { name: timezoneName } = this.getSelectedTimezone();
		return getFormattedSelectedDates(this.selectedDateState, this.mode, timezoneName);
	};

	private onDropdownChange = ({ detail: isDropdownOpen }: CustomEvent<boolean>) => {
		this.dropdownOpen = isDropdownOpen;
		this.dropdownStateChange.emit(isDropdownOpen);
	};

	private getTextFieldTooltip = (): string | undefined => {
		if (this.selectedDates?.length > 0) {
			return buildTooltipText(this.selectedDates, this.getSelectedTimezone(), this.timezones);
		}
	};

	private getDropdownInputValue = (): string | undefined => {
		return buildCustomIntervalTimeRange(this.selectedDates, this.getSelectedTimezone().name);
	};

	private getInputConfig = (): Partial<ITextField> => {
		const inputValue = this.getDropdownInputValue();

		return merge({}, DEFAULT_TIME_RANGE_PICKER_INPUT_CONFIG, this.inputConfig, {
			value: inputValue,
			tooltipConfig: { text: this.getTextFieldTooltip(), position: this.tooltipPosition }
		});
	};

	private onClickApply = () => {
		this.selectedDatesChange.emit(this.selectedDateState);
		this.dropdownStateChange.emit(false);
		this.dropdownOpen = false;
	};

	private onClickCancel = (event: CustomEvent<MouseEvent>) => {
		this.undoLastChanges();
		this.cancelClicked.emit(event);
		this.dropdownOpen = false;
	};

	private undoLastChanges = () => {
		if (!isEmpty(this.selectedDates)) {
			this.selectedDateState = this.selectedDates;
		} else {
			this.selectedDateState = [];
		}
	};

	private isApplyButtonDisabled() {
		if (isEmpty(this.selectedDateState)) return true;

		return !hasRangeChanged(this.selectedDateState, this.selectedDates);
	}

	render() {
		const dropdownPositionConfig = this.dropdownPositionOptions;
		const inputConfig = this.getInputConfig();

		return (
			<Host>
				<kv-dropdown isOpen={this.dropdownOpen} onOpenStateChange={this.onDropdownChange} inputConfig={inputConfig} options={dropdownPositionConfig}>
					<div class="absolute-time-content">
						<kv-absolute-time-picker
							mode={this.mode}
							headerTitle={this.headerTitle}
							selectedDates={this.getAbsoluteRange()}
							disabledDates={this.disabledDates}
							initialDate={this.calendarInitialDate}
							onSelectedDatesChange={this.handleAbsoluteDatesChange}
							calendarInputMaxDate={this.calendarInputMaxDate}
							calendarInputMinDate={this.calendarInputMinDate}
						/>
						<div class="footer">
							<div class="actions">
								<kv-action-button-text type={EActionButtonType.Tertiary} size={EComponentSize.Small} text="Cancel" onClickButton={this.onClickCancel} />
								<kv-action-button-text
									type={EActionButtonType.Primary}
									size={EComponentSize.Small}
									text="Apply"
									disabled={this.isApplyButtonDisabled()}
									onClickButton={this.onClickApply}
								/>
							</div>
						</div>
					</div>
				</kv-dropdown>
			</Host>
		);
	}
}
