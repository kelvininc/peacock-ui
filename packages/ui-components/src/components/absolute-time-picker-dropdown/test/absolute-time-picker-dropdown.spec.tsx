import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvAbsoluteTimePickerDropdown } from '../absolute-time-picker-dropdown';
import { h } from '@stencil/core';

describe('Absolute Time Picker Dropdown (unit tests)', () => {
	let page: SpecPage;

	describe('when default props and initial date are used', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvAbsoluteTimePickerDropdown],
				template: () => <kv-absolute-time-picker-dropdown initialDate="2023-03-03" />
			});
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});

	describe('when a selectedDate is provided', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvAbsoluteTimePickerDropdown],
				template: () => <kv-absolute-time-picker-dropdown selectedDates={[1681319856833]} />
			});
		});

		it('should match the snapshot, the initial date on the calendar should be inherited from the selected date ("2023-04-12")', () => {
			expect(page.root).toMatchSnapshot();
		});
	});
});
