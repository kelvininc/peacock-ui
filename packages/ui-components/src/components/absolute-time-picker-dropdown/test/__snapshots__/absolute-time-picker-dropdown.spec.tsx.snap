// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Absolute Time Picker Dropdown (unit tests) when a selectedDate is provided should match the snapshot, the initial date on the calendar should be inherited from the selected date ("2023-04-12") 1`] = `
<kv-absolute-time-picker-dropdown>
  <kv-dropdown>
    <div class="absolute-time-content">
      <kv-absolute-time-picker headertitle="" initialdate="2023-04-12" mode="single"></kv-absolute-time-picker>
      <div class="footer">
        <div class="actions">
          <kv-action-button-text size="small" text="Cancel" type="tertiary"></kv-action-button-text>
          <kv-action-button-text disabled="" size="small" text="Apply" type="primary"></kv-action-button-text>
        </div>
      </div>
    </div>
  </kv-dropdown>
</kv-absolute-time-picker-dropdown>
`;

exports[`Absolute Time Picker Dropdown (unit tests) when default props and initial date are used should match the snapshot 1`] = `
<kv-absolute-time-picker-dropdown>
  <kv-dropdown>
    <div class="absolute-time-content">
      <kv-absolute-time-picker headertitle="" initialdate="2023-03-03" mode="single"></kv-absolute-time-picker>
      <div class="footer">
        <div class="actions">
          <kv-action-button-text size="small" text="Cancel" type="tertiary"></kv-action-button-text>
          <kv-action-button-text disabled="" size="small" text="Apply" type="primary"></kv-action-button-text>
        </div>
      </div>
    </div>
  </kv-dropdown>
</kv-absolute-time-picker-dropdown>
`;
